src/bin/pg_autoctl/pgsetup.c

```diff
@@ -1886,16 +1905,22 @@ pg_setup_standby_slot_supported(PostgresSetup *pgSetup, int logLevel)
 		return false;
 	}
 
+	/* Postgres 12.4 and up are good */
+	if (pg_version >= 1204 && pg_version < 1300)
+	{
+		return true;
+	}
+
 	/* Starting with Postgres 13, all versions are known to have the bug fix */
-	if (pgSetup->control.pg_control_version >= 1300)
+	if (pg_version >= 1300)
 	{
 		return true;
 	}
 
 	/* should not happen */
 	log_debug("BUG in pg_setup_standby_slot_supported(%d): "
```

Comment:
Maybe make this `log_warn`, so that people will report this bug quicker?
```suggestion
	log_warn("BUG in pg_setup_standby_slot_supported(%d): "
```

Reply:
All the other BUG messages are log_debug, so that's what I'm using here. Also we use that function at every loop in the keeper, so that's going to be a lot of warning messages, and the users can't do anything about it (other than all report the same bug). Maybe we could add a warning at startup?

Reply:
Yeah logging this every loop seems even worse. If it's easy to do at startup (1-3 line changes) I'd say let's do it. Otherwise I don't think it's important enough.

Reply:
On the other hand we already have `pg_autoctl config check` to implement an explicit check and warning about that case. When we suspect something is wrong with replication slots maintenance on the standby nodes we can always ask to run the check command.

```
$ pg_autoctl config check --pgdata node1
11:28:39 23550 INFO  Postgres setup for PGDATA "/Users/dim/dev/MS/pg_auto_failover/tmux/node1" is ok, running with PID 5501 and port 23446
11:28:39 23550 INFO  Connection to local Postgres ok, using "port=5501 dbname=demo host=/tmp"
11:28:39 23550 INFO  Postgres configuration settings required for pg_auto_failover are ok
11:28:39 23550 INFO  Postgres version "12.6" allows using replication slots on the standby nodes
...
```